map/bookmark_manager.cpp

```diff
@@ -404,6 +404,59 @@ void BookmarkManager::ResetRecentlyDeletedBookmark()
   m_recentlyDeletedBookmark.reset();
 }
 
+size_t BookmarkManager::GetRecentlyDeletedCategoriesCount() const
+{
+  Platform::FilesList files;
+  Platform::GetFilesByExt(GetTrashDirectory(), kKmlExtension, files);
+  return files.size();
+}
+
+BookmarkManager::KMLDataCollectionPtr BookmarkManager::GetRecentlyDeletedCategories()
+{
+  auto collection = LoadBookmarks(GetTrashDirectory(), kKmlExtension, KmlFileType::Text,
+                                  [](kml::FileData const &)
+                                  {
+    return true;
+  });
+  return collection;
+}
+
+bool BookmarkManager::IsRecentlyDeletedCategory(const std::string &filePath) const
+{
+  Platform::FilesList files;
+  Platform::GetFilesByExt(GetTrashDirectory(), kKmlExtension, files);
+  auto const it = std::find(files.begin(), files.end(), base::FileNameFromFullPath(filePath));
+  return it != files.end();
```

Comment:
Will this work?
```suggestion
  return filePath.find(GetTrashDirectory()) != std::string::npos;
```

Reply:
Thanks! Semantically it is not the scanning the folder but it works as expected (and faster).